src/commands/periscope/helpers/periscopehelper.ts

```diff
@@ -214,6 +216,105 @@ export async function generateDownloadableLinks(
     }
 }
 
+export async function downloadableLinks(
+    periscopeStorage: PeriscopeStorage,
+    output: string,
+    nodeName: string
+): Promise<any> {
+
+    try {
+        const storageAccount = periscopeStorage.storageName;
+        const storageKey = periscopeStorage.storageKey;
+
+        // Get DNS Core hostname which Periscope use it as name of the container.
+        const matches = output.match(/(https?:\/\/[^\s]+)/g);
+        let containerName;
+        if (matches![0].indexOf('://') !== -1) {
+            containerName = matches![0].replace('https://', '').split('.')[0];
+        }
+
+        // Use SharedKeyCredential with storage account and account key
+        const sharedKeyCredential = new StorageSharedKeyCredential(storageAccount, storageKey);
+        const blobServiceClient = new BlobServiceClient(
+            `https://${storageAccount}.blob.core.windows.net`,
+            sharedKeyCredential
+        );
+
+        // Hide this all under single pupose funct which lik elike : getZipDir or get me dirname
+        const containerClient = blobServiceClient.getContainerClient(containerName);
+
+        // List all current blob.
+        const listCurrentUploadedFolders = [];
+        for await (const item of containerClient.listBlobsByHierarchy("/")) {
+            if (item.kind === "prefix") {
+                listCurrentUploadedFolders.push(item.name);
+            }
+        }
+
+        // Sort and get the latest uploaded folder under the container used by periscope.
+        // const periscopeHtmlInterfaceList = [];
+        // jszip for zipping
+        const zip = new JSZip();
+
+        for await (const blob of containerClient.listBlobsFlat()) {
+            // Get the latest uploaded folder, then Identify the Zip files in the latest uploaded logs within that folder
+            // and extract *.zip files which are individual node logs.
+            const latestBlobUploadedByPeriscope = blob.name.indexOf(listCurrentUploadedFolders.sort().reverse()[0]);
```

Comment:
This naming is confusing.  It clearly isn't a blob - it appears to be an index into a string.  Also it appears to be used only for whether the substring is present or not, not the index itself.  Perhaps consider:

```suggestion
            const latestFolder = listCurrentUploadedFolders.sort().reverse()[0];
            const blobMatchesLatestFolder= blob.name.includes(latestFolder);
```

_If_ I have understood this correctly, of course... big if!